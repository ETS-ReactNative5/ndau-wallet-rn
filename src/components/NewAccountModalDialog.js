import React, { Component } from 'react'

import { StyleSheet, Text, View } from 'react-native'
import {
  widthPercentageToDP as wp,
  heightPercentageToDP as hp
} from 'react-native-responsive-screen'
import ModalDialog from './ModalDialog'
import CommonButton from '../components/CommonButton'
import cssStyles from '../css/styles'
import NumberPicker from './NumberPicker'

class NewAccountModalDialog extends Component {
  addNewAccount = () => {
    this.props.setModalVisible(false)
    this.props.addNewAccount()
    this.props.closeModal()
  }

  render () {
    const innerViewStyle = {
      height: hp('40%')
    }
    return (
      <ModalDialog innerViewStyle={innerViewStyle} {...this.props}>
        <Text style={styles.text}>
          How many new accounts would you like to add to your wallet?
        </Text>
        <NumberPicker {...this.props} />
        <View style={cssStyles.footer}>
          <CommonButton onPress={this.addNewAccount} title='Add new' />
        </View>
      </ModalDialog>
    )
  }
}

var styles = StyleSheet.create({
  text: {
    color: '#ffffff',
    fontSize: 20,
    fontFamily: 'TitilliumWeb-Regular',
    marginTop: hp('1%'),
    marginBottom: hp('1%'),
    marginLeft: wp('1%'),
    marginRight: wp('1%')
  }
})

export default NewAccountModalDialog
