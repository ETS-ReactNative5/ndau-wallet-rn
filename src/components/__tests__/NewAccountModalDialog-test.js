import { mount } from 'enzyme'
import React from 'react'
import { StyleSheet } from 'react-native'
import NewAccountModalDialog from '../NewAccountModalDialog'

import renderer from 'react-test-renderer'

describe('testing NewAccountModalDialog...', () => {
  let styles = StyleSheet.create({
    wizardText: {
      color: '#ffffff',
      fontSize: 20
    }
  })
  const navigator = {
    setStyle: () => {}
  }

  it('renders correctly', () => {
    const tree = renderer
      .create(
        <NewAccountModalDialog parentStyles={styles} navigator={navigator} />
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
