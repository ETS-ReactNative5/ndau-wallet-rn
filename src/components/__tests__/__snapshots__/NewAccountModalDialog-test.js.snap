// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing NewAccountModalDialog... renders correctly 1`] = `
<Modal
  animationType="slide"
  hardwareAccelerated={false}
  innerViewStyle={
    Object {
      "height": 533.5,
    }
  }
  navigator={
    Object {
      "setStyle": [Function],
    }
  }
  onRequestClose={[Function]}
  parentStyles={
    Object {
      "wizardText": Object {
        "color": "#ffffff",
        "fontSize": 20,
      },
    }
  }
  transparent={true}
  visible={true}
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "column",
          "justifyContent": "center",
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#333333",
            "borderRadius": 3,
            "height": 934,
            "paddingBottom": 26.5,
            "paddingLeft": 15,
            "paddingRight": 15,
            "width": 675,
          },
          Object {
            "height": 533.5,
          },
        ]
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={null}
      >
        <Text
          accessible={true}
          allowFontScaling={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#A9A9A9",
                "fontSize": 20,
              },
              Object {
                "marginLeft": 7.5,
                "marginTop": 6.5,
              },
              Object {
                "fontFamily": "FontAwesome",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
            ]
          }
        >
          
        </Text>
      </View>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#ffffff",
            "fontFamily": "TitilliumWeb-Regular",
            "fontSize": 20,
            "marginBottom": 13.5,
            "marginLeft": 7.5,
            "marginRight": 7.5,
            "marginTop": 13.5,
          }
        }
      >
        How many new accounts would you like to add to your wallet?
      </Text>
      <View
        style={
          Object {
            "alignContent": "center",
            "flex": 1,
            "justifyContent": "center",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "center",
            }
          }
        >
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "#A9A9A9",
                    "fontSize": 32,
                  },
                  undefined,
                  Object {
                    "fontFamily": "FontAwesome",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                ]
              }
            >
              
            </Text>
          </View>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#ffffff",
                "fontFamily": "TitilliumWeb-Regular",
                "fontSize": 32,
                "marginLeft": 75,
                "marginRight": 75,
              }
            }
          />
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "#A9A9A9",
                    "fontSize": 32,
                  },
                  undefined,
                  Object {
                    "fontFamily": "FontAwesome",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                ]
              }
            >
              
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "justifyContent": "flex-end",
          }
        }
      >
        <View
          accessibilityComponentType="button"
          accessibilityTraits="button"
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignContent": "center",
              "backgroundColor": "#4e957a",
              "borderRadius": 3,
              "height": 93.5,
              "justifyContent": "center",
              "opacity": 1,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "alignSelf": "center",
                "color": "#ffffff",
                "fontFamily": "TitilliumWeb-Regular",
                "fontSize": 16,
                "fontWeight": "bold",
              }
            }
          >
            Add new
          </Text>
        </View>
      </View>
    </View>
  </View>
</Modal>
`;
